pilot/pkg/model/context.go

```diff
@@ -205,8 +205,14 @@ func (node *Proxy) GetRouterMode() RouterMode {
 // Listener generation code will still use the SidecarScope object directly
 // as it needs the set of services for each listener port.
 func (node *Proxy) SetSidecarScope(ps *PushContext) {
-	labels := node.WorkloadLabels
-	node.SidecarScope = ps.getSidecarScope(node, labels)
+	if node.Type == SidecarProxy {
+		labels := node.WorkloadLabels
+		node.SidecarScope = ps.getSidecarScope(node, labels)
+	} else {
+		// Gateways should just have a default scope with egress: */*
```

Comment:
Not sure - Gateway has now a way to specify an explicit set of namespaces. 
Maybe ok for now ( not changing behavior ), but we should add a TODO: to scan if the Gateway 
is using only "namespace/foo" for redirection that this is used as equivalent with the Sidecar, i.e. 
ignore any namespace not explicitly referenced by Gateway.

Reply:
@mandarjog had a PR a while back about scoping gateways. Not sure what happened to it. Maybe this would be relevant for that